docs/tutorial-node-implementation.md

```diff
@@ -0,0 +1,559 @@
+    cmd = ['python3', '-m', 'poetry', 'export', '--without-hashes']
+    requirments = subprocess.run(cmd, stdout=PIPE)
+    req_file = Path('requirements.txt')
+    with req_file.open(mode='w') as f:
+        f.write(f'{target.resolve()}\n')
+        f.write(requirments.stdout.decode())
+    cmd = ['python3', '-m', 'pip', 'install', '-r', str(req_file.resolve())]
+    print(cmd_join(cmd))
+    subprocess.run(cmd)
+
+
+def install_plugin(zip_da):
+    if not zip_da.exists():
+        print('ERROR: DA not found at ', zip_da)
+        return
+    extract_zip(zip_da, Path('plugins'))
+
+
+def post_install(environ):
+    extra_env = {
+        'FLASK_APP': 'qhana_plugin_runner',
+        'FLASK_ENV': 'production',
+        'PLUGIN_FOLDERS': 'plugins:git-plugins'
+    }
+
+    cmd = ['python3', '-m', 'invoke', 'load-git-plugins']
+    print(cmd_join(cmd), 'env:', extra_env)
+    subprocess.run(cmd, env=ChainMap(extra_env, environ))
+    cmd = ['python3', '-m', 'flask', 'install']
+    print(cmd_join(cmd), 'env:', extra_env)
+    subprocess.run(cmd, env=ChainMap(extra_env, environ))
+
+
+def main(argv):
+    print('pwd', Path('.').resolve())
+    print(argv)
+    env = ChainMap(argv_to_dict(argv), environ)
+    install_requirements(env)
+    install_deployment_artifacts(env)
+    post_install(env)
+
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+### Explanation
+
+`PythonScriptArtifacts` get the user input not as environment variables but as command-line arguments.
+The function `argv_to_dict` converts the command-line arguments to a dictionary for easy access.
+The `ChainMap` class is used to merge multiple dictionaries, i.e., to merge the command-line arguments with the environment variables.
+`install_requirements` installs git.
+The file names of the `Deployment Artifacts` are passed as environment variables to the script under the key `DAs`, and the path to the unzipped `CSAR` folder can be found under the key `CSAR`.
+The function `install_deployment_artifacts` calculates the paths to the `Deployment Artifacts`,
+checks the type, and calls the correct install functions.
+`install_plugin_runner` unpacks the passed `Deployment Artifact` and installs required Python packages, `install_plugin` unpacks the passed `Deployment Artifact` into the plugins folder.
+In the end, `post_install` executes commands to configure the application and install more dependencies.
+
+
+## Deployment Artifacts
+
+`Deployment Artifacts` implement the business logic of an application, e.g., the JAR-file for a Tomcat web server.
+If you want to create a `Node Type` that has, e.g., a JAR-file as a `Deployment Artifact`, you need to create an interface with an implementation artifact that can execute this file.
+An environment variable called `DAs` is passed to `operations` and contains a list of all `Deployment Artifacts` of this `Node Type` and `Node Template`.
+
+**Example with a JAR-file as deployment artifact**
```

Comment:
Link the original explanations and shortly explain what this is because this will confuse readers if there ios a different Node Type / Node Type Implementation suddenly.